spot-client/src/spot-tv/ui/loaders/withCalendar.js

```diff
@@ -104,13 +114,23 @@ export class CalendarLoader extends AbstractLoader {
 
         return calendarService.initialize(this.props.calendarType)
             .then(() => {
-                if (isSetupComplete && this.props.calendarEmail) {
-                    calendarService.startPollingForEvents({
-                        email: this.props.calendarEmail,
-                        jwt: this.props.jwt
-                    });
-                }
+                this.maybeStartPolling();
+            });
+    }
+
+    /**
+     * Starts the events polling if all underlying conditions have been met.
+     *
+     * @returns {void}
+     */
+    maybeStartPolling() {
+        if (this.props.isSetupComplete && this.state.loaded && !this.state.isPollingStarted) {
+            calendarService.startPollingForEvents({
+                email: this.props.calendarEmail,
+                jwt: this.props.jwt
             });
+            this.setState({ isPollingStarted: true });
```

Comment:
Do you need this flag? What if the service could no-op if already polling?

Reply:
yeah it is already doing no-op